Emagging drones will now invert their third law (and ONLY their third law)

Ricotez opened the pull request:
When you emag a drone, you will turn its third law into this:

<b>3. Your goals are to destroy, sabotage, break, tear apart, and depower to the best of your abilities, You must never actively work against these goals.</b>

Drones also have a new var, can_emag, that decides whether they can be emagged. Emagging a drone sets it to 0 (obviously), but syndrones also have it set to 0 because why would you want to emag a syndrone.

Based on this minor suggestion: https://tgstation13.org/phpBB/viewtopic.php?f=9&t=30&start=1000#p92177

<b>UPDATE 1: Makes emagged drones more obvious and adds penalties.</b>
Emagged drones will now jitter, which means they are visibly shaking, and they have an additional examine message: "Its movements are jittery and it appears to be falling apart."
Also, every time handle_regular_status_updates() is called on the drone, it has a 15% chance of losing a health point and visibly sparking, which means that emagged drones will eventually die.

Commit:
-Adds a new var to drones, can_emag, and an implementation for emag_act.
-Emag_act will edit the drone's laws, but ONLY the third law, instructing it to deconstruct the station. The first and second law remain the same as before.
-Syndrones have can_emag set to 0 and cannot be emagged.

Steelpoint commented:
I highly disagree with this :-1: 

Drones are one of the easiest ways right now to get back into the game, and this change would turn it into a free antag position. Trust me when I say that a antag Drone can cause a lot of damage, looting the entire armoury within moments and "coincidently" opening up the brig is only the tip of the iceberg compared to the damage it could do.

Keep it for a admin event or trigger, but not something anyone with a emag can do.

Razharas commented:
This is absolutely OP

PKPenguin321 commented:
goof tried this and got shut down by collaborators did he not
:-1:

Actionb commented:
Please no. Leave drones as the silent, neutral helpers. Don't make them a target.

Commit:
-Emagged drones will now jitter and have an extra examine message, making them obvious.
-In addition, emagged drones will occasionally spark and lose 1 health. There's a 15% chance of this happening with every status update.

Ricotez commented:
<b>UPDATE 1: Makes emagged drones more obvious and adds penalties.</b>
Emagged drones will now jitter, visibly shaking, and they have an additional examine message: "Its movements are jittery and it appears to be falling apart."
Also, every time handle_regular_status_updates() is called on the drone, it has a 15% chance of losing a health point and visibly sparking, which means that emagged drones will eventually die.

Steelpoint commented:
I'm sorry but this is still a bad idea, its too powerful and easy to abuse.

Make it a admin event or some obscure and expensive AI power, but not something a cheap and common EMAG can do.

nukeop commented:
Roboticist with an emag could wreak absolute havoc and would be nearly undetectable.

Ricotez commented:
if all drones leave robotics jittering and sparking I think it would be extremely obvious what's going on

Steelpoint commented:
A jittering Drone can still cause massive damage to the station, in addition this will both incentivise Drones to loiter around until someone accidently swipes a Emag on them as well as have the crew be very vigilant on killing Drones on sight 'just to be sure'.

This could also get the Roboticist lynced whenever someone emags a drone.

Ricotez commented:
@Steelpoint you raise good points, but I'll wait for more feedback before I decide whether I'm gonna close this or not

optionally I could also change "emagged" into "corrupted" and link it to something else, such as an AI power like you suggested. but in that case I should probably get rid of the penalties.

Tokiko1 commented:
Consider restricting the emagging to the shell or even the fabricator. That way random people in the hallways can't just emag any drone they see. That would also stop people random destroying functional drones.

freerealestate commented:
Your first update made them pretty shitty to play, but I think it balanced
them well.

On 29 June 2015 at 01:01, Tokiko1 <notifications@github.com> wrote:

> Consider restricting the emagging to the shell or even the fabricator.
> That way random people in the hallways can't just emag any drone they see.
> That would also stop people random destroying functional drones.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/10227#issuecomment-116275601>
> .
>

PKPenguin321 commented:
:-1: ghosts should not get free unlimited antag slots, drones should remain a neutral third party

Cheridan commented:
I agree that drones should remain friendly neutral helpers.

Ricotez commented:
God has spoken.
